internal/extproc/translator/openai_gcpvertexai.go

```diff
@@ -49,10 +59,89 @@ func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseHeaders(headers
 
 // ResponseBody implements [Translator.ResponseBody] for GCP Gemini.
 // This method translates a GCP Gemini API response to the OpenAI ChatCompletion format.
-func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseBody(respHeaders map[string]string, body io.Reader, endOfStream bool) (
+func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseBody(_ map[string]string, body io.Reader, _ bool) (
 	headerMutation *extprocv3.HeaderMutation, bodyMutation *extprocv3.BodyMutation, tokenUsage LLMTokenUsage, err error,
 ) {
-	// TODO: Implement response body translation from GCP Gemini to OpenAI format
-	_, _, _ = respHeaders, body, endOfStream
-	return nil, nil, LLMTokenUsage{}, nil
+	// Read the body
+	bodyBytes, err := io.ReadAll(body)
+	if err != nil {
+		return nil, nil, LLMTokenUsage{}, fmt.Errorf("error reading response body: %w", err)
+	}
```

Comment:
this unnecessarily copies the body. You can directly unmarshal into the struct via `json.NewDecoder(body).Decode`.